src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -133,53 +122,120 @@ public String toString() {
     return summary();
   }
 
-  public void setStepSizeForCache(final long stepSizeInBytes) {
-    this.stepSizeInBytes.put(cacheType, stepSizeInBytes);
-  }
-
-  public Long getCurrentCacheMaxSizeInBytes() {
+  public long getCurrentCacheMaxSizeInBytes() {
     return currentCacheMaxSizeInBytes;
   }
 
-  public Long getDesiredCacheMaxSizeInBytes() {
+  public long getDesiredCacheMaxSizeInBytes() {
     return desiredCacheMaxSizeInBytes;
   }
 
   public ResourceEnum getCacheType() {
     return cacheType;
   }
 
-  private boolean validateAndSetConfigValues() {
-    // This is intentionally not made static because different nodes can
-    // have different bounds based on instance types
+  public static final class Builder {
+    public static final long DEFAULT_COOL_OFF_PERIOD_IN_MILLIS = 300 * 1_000;
+    public static final boolean DEFAULT_IS_INCREASE = true;
+    public static final boolean DEFAULT_CAN_UPDATE = true;
+    // TODO: Update the step size to also include percentage of heap size along with absolute value
+    public static final ImmutableMap<ResourceEnum, Long> DEFAULT_STEP_SIZE_IN_BYTES =
+        new ImmutableMap.Builder<ResourceEnum, Long>()
+            .put(
+                ResourceEnum.FIELD_DATA_CACHE,
+                512 * 1_024L * 1_024L) // Field data cache having step size of 512MB
+            .put(
+                ResourceEnum.SHARD_REQUEST_CACHE,
+                512 * 1_024L) // Shard request cache step size of 512KB
+            .build();
+
+    private final ResourceEnum cacheType;
+    private final NodeKey esNode;
+    private final AppContext appContext;
+    private double upperBoundThreshold;
+
+    private long stepSize;
+    private boolean isIncrease;
+    private boolean canUpdate;
+    private long coolOffPeriodInMillis;
+
+    private Long currentCacheMaxSizeInBytes;
+    private Long desiredCacheMaxSizeInBytes;
+    private Long heapMaxSizeInBytes;
+
+    public Builder(
+        final NodeKey esNode,
+        final ResourceEnum cacheType,
+        final AppContext appContext,
+        final double upperBoundThreshold) {
```

Comment:
we should read upper bound / lower bound from RcaConf. Let's use a switch to assign it to default for now

Reply:
Presently the cache decider is reading from rcaConf. It parses and sends the required fields to actions. It makes sense to retain this. Aditya has a PR open that makes changes to deciderConfig structure. After which there will be a modification in this.